### PR TITLE
[azure] Refactor client config

### PR DESCRIPTION
**Pull request by paulmey**
Refactoring configuration and creation of the Azure client. Preparing to add a chroot builder for Azure.

### CODE REVIEW

**Comment by paulmey** [comment]
@joellopes03, please take a look and see that this doesn't re-trigger the MSI bug. I tested with MSI, but better to have more eyes on it.

**Comment by paulmey** [comment]
@sumit-kalra, please review, thanks!

File: builder/azure/arm/builder.go [hunk 1]
```diff
@@ -57,13 +57,10 @@ func (b *Builder) Run(ctx context.Context, ui packer.Ui, hook packer.Hook) (pack
...
 	defer cancel()
 
-	// User's intent to use MSI is indicated with empty subscription id, tenant, client id, client cert, client secret and jwt.
-	// FillParameters function will set subscription and tenant id here. Therefore getServicePrincipalTokens won't select right auth type.
-	// If we run this after getServicePrincipalTokens call then getServicePrincipalTokens won't have tenant id.
-	if !b.config.useMSI() {
-		if err := newConfigRetriever().FillParameters(b.config); err != nil {
-			return nil, err
-		}
+	// FillParameters function will set the authType from supplied parameters set the subscription and tenant id.
```
Comment (sumit-kalra) [inline builder/azure/arm/builder.go]: NIT: Missing comma after "supplied parameters"? Please consider revising this sentence.

**Comment by paulmey** [comment]
@azr or @SwampDragons, can you PTAL if I did the struct comments right for this change?

**Pull request merged**